.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "windows/mdns"]
```

Comment:
Did you try with Conan? I have really bad experience with Git submodules

Reply:
I spent some time looking into it - it seems to be more of a CID tool which generates binaries? I could not figure out how to integrate it into either Android Studio or Visual Studio? all we really need is to pull the mdns.h file from the repo

Reply:
I haven't used it myself, but just briefly reading the getting started guide, it looked to me like Maven.
But well, I can try to integrate it at some point.

Reply:
I think this requires further study. How does the pub.dev deployment work? From what I can tell is that when you type flutter build windows, it creates a visual studio project and copies all the sources into it, including the sources out of the plugin's windows folder. So all the sources need to be in there. We certainly do not want that the plugin user needs to install the conan client just to get all the sources.

Reply:
Good point. I don't know how pub.dev deployment works. We'll have to try and see :)

Reply:
OK the way it works right now with the submodules is that if you clone the repo, the mdns sources are not included. One needs to run git submodule update to pull in the sources.

I did a flutter pub publish --dry-run and this is what would be published from the windows folder:

'-- windows
    |-- CMakeLists.txt
    |-- flutter_nsd_plugin.cpp
    |-- include
    |   '-- flutter_nsd
    |       '-- flutter_nsd_plugin.h
    |-- mdns
    |   |-- CHANGELOG
    |   |-- CMakeLists.txt
    |   |-- LICENSE
    |   |-- README.md
    |   |-- cmake
    |   |   '-- Config.cmake.in
    |   |-- mdns.c
    |   '-- mdns.h
    |-- mdns_impl.c
    '-- mdns_impl.h

as you can see, the entire mdns repo will be included, not just the mdns.h file. Not ideal. But the only way to fix that would be to just manually copy/paste the file

the guidelines say:
Your package should depend only on hosted dependencies (from the default pub package server) and SDK dependencies (sdk: flutter). These restrictions ensure that dependencies of your packages cannot become unavailable in the future. So we need to include the mdns.h file in the package, and not just a dependency.

Reply:
So I would suggest that I remove the sub-package dependency, then copy the mdns.h file manually (just as I did in the first version), then add a README.md into that folder that tells where that file came from and how to update it if needed.

Reply:
I had a look into another Flutter plugin with Windows support and I see that they're including the library dependencies in binary format (either lib/dll). Actually not only that they include dependencies in binary format, but also the C++ code itself. The only visible code are the Flutter plugin cpp and h files. You can see it here for example: https://github.com/dynamsoftlabs/flutter_barcode_sdk/tree/main/windows

So basically, we'll need all the C++ code to be in binary format and their header files and `flutter_nsd_plugin.cpp` + `flutter_nsd_plugin.h`.

Reply:
Another Flutter plugin with Windows support that follows the same pattern: binaries + headers + own plugin cpp/h. https://github.com/jerson/flutter-openpgp/tree/master/windows

Reply:
Yes I can see that. However it would depend on how these libraries are distributed originally. The mdns lib is just in a .h file so it may be a bit tricky to turn that into a binary. One could compile the mdns_impl.c however that is part of the plugin source. either way I tried building a project with the sources included and it compiled just fine.

Reply:
Can we rename the mdns.h to .cpp and extract a header file for it so that we can compile it and include it as a binary?

Reply:
I don't think it is a rule to include binaries. This may apply if those libraries are not available as source. But in this case where the lib is distributed as source it makes more sense to include the source. The guidelines only say that everything which is required to build the plugin should be included in the package so that it does not rely on external repos which may not be there in the future.
And this mdns.h is explicitly labeled as public domain

Reply:
It's true that it's not a rule, but it's also not part of the plugin's source code that we maintain therefore I believe it shouldn't be part of the plugin's Git repository. Maybe I'm just used to Android dev too much :)
Ideally, I'd prefer to have a dependency management system like Cocoapods for MacOS/iOS and Gradle for Android, but looks like Flutter doesn't support any out of the box.

It would be great if we could only include the binaries so that there's a clear separation between plugin code and library code,  but I understand that it might require extra time to get working. If you prefer not to spend more time on this, then we can merge it and I'll look into it at some point. Thanks!